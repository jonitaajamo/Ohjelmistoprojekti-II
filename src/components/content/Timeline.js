import React from "react";

const Timeline = () => (
  <article className="tile is-child notification is-danger">
    <input className="progress" type="range"></input>
  </article>
);

export default Timeline;
