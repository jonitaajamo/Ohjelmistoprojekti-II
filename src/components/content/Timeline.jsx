import React from "react";
import TimelineTabs from "./TimelineTabs";

const Timeline = () => (
  <article className="tile is-child notification">
  <TimelineTabs />
    <input className="progress" type="range"></input>
  </article>
);

export default Timeline;
