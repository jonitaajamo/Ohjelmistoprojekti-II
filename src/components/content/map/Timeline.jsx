import React, { Component } from "react";
import Slider from "rc-slider";
import "rc-slider/assets/index.css";
import Tabs from "./timeline/Tabs";

export default class Timeline extends Component {
  renderDatalist() {
    const marksObject = {};
    for (
      let index = 0;
      index < this.props.geographicalWeightData.length;
      index++
    ) {
      if (index % 6 === 0) {
        marksObject[index] = this.props.geographicalWeightData[index].month;
      }
    }
    return marksObject;
  }

  render() {
    const timelineStyle = {
      alignItems: "center",
      justifyContent: "center"
    };

    return (
      <article className="tile is-child notification">
        <Tabs
          geographicalWeightData={this.props.geographicalWeightData}
          onAssetChange={this.props.onAssetChange}
        />
        <div style={timelineStyle}>
          <Slider
            marks={this.renderDatalist()}
            step={1}
            min={0}
            max={this.props.length}
            defaultValue={this.props.month}
            dotStyle={{ borderColor: "orange" }}
            activeDotStyle={{ borderColor: "yellow" }}
            onChange={this.props.onChange}
          />
        </div>
        <br />
        <span className="field has-addons has-addons-centered">
          <a
            className="button left"
            id="previous"
            onClick={this.props.onMonthButtonClick}
          >
            <span className="icon left">
              <i className="fas fa-arrow-left left" />
            </span>
            <span className="left">Previous month</span>
          </a>
          <a className="button left">
            <span className="icon left">
              <i className="fas fa-play left" />
            </span>
          </a>
          <a
            className="button right"
            id="next"
            onClick={this.props.onMonthButtonClick}
          >
            <span className="right">Next month</span>
            <span className="icon right">
              <i className="fas fa-arrow-right right" />
            </span>
          </a>
        </span>
      </article>
    );
  }
}
