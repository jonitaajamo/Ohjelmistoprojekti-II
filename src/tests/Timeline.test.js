import React from "react";
import Timeline from "../components/content/Timeline";

describe("Timeline", () => {
  const props = {
    length: 76,
    month: 10
  };

  it("renders correctly", () => {
    const wrapper = shallow(<Timeline {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders child", () => {
    const wrapper = shallow(<Timeline {...props} />);
    expect(wrapper.find("TimelineTabs").length).toEqual(1);
  });

  it("sets props correctly", () => {
    const wrapper = shallow(<Timeline {...props} />);
    expect(wrapper.find("input").props().max).toBe(76);
    expect(wrapper.find("input").props().value).toBe(10);
  });

  it("calls onChange when changed", () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Timeline {...props} onChange={onChange} />);
    wrapper.find("input").simulate("change");
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
