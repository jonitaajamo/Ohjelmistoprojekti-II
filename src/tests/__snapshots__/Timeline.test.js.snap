// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Timeline
    length={76}
    month={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TimelineTabs
          geographicalWeightData={undefined}
/>,
        <input
          className="progress"
          max={76}
          min="0"
          onChange={undefined}
          type="range"
          value={10}
/>,
        <span
          className="field has-addons has-addons-centered"
>
          <a
                    className="button left"
                    id="previous"
                    onClick={undefined}
          >
                    <span
                              className="icon left"
                    >
                              <i
                                        className="fas fa-arrow-left left"
                              />
                    </span>
                    <span
                              className="left"
                    >
                              Previous month
                    </span>
          </a>
          <a
                    className="button right"
                    id="next"
                    onClick={undefined}
          >
                    <span
                              className="icon right"
                    >
                              <i
                                        className="fas fa-arrow-right right"
                              />
                    </span>
                    <span
                              className="right"
                    >
                              Next month
                    </span>
          </a>
</span>,
      ],
      "className": "tile is-child notification",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "geographicalWeightData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "progress",
          "max": 76,
          "min": "0",
          "onChange": undefined,
          "type": "range",
          "value": 10,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="button left"
              id="previous"
              onClick={undefined}
>
              <span
                            className="icon left"
              >
                            <i
                                          className="fas fa-arrow-left left"
                            />
              </span>
              <span
                            className="left"
              >
                            Previous month
              </span>
</a>,
            <a
              className="button right"
              id="next"
              onClick={undefined}
>
              <span
                            className="icon right"
              >
                            <i
                                          className="fas fa-arrow-right right"
                            />
              </span>
              <span
                            className="right"
              >
                            Next month
              </span>
</a>,
          ],
          "className": "field has-addons has-addons-centered",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="icon left"
>
                  <i
                                    className="fas fa-arrow-left left"
                  />
</span>,
                <span
                  className="left"
>
                  Previous month
</span>,
              ],
              "className": "button left",
              "id": "previous",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fas fa-arrow-left left"
/>,
                  "className": "icon left",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-arrow-left left",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Previous month",
                  "className": "left",
                },
                "ref": null,
                "rendered": "Previous month",
                "type": "span",
              },
            ],
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="icon right"
>
                  <i
                                    className="fas fa-arrow-right right"
                  />
</span>,
                <span
                  className="right"
>
                  Next month
</span>,
              ],
              "className": "button right",
              "id": "next",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="fas fa-arrow-right right"
/>,
                  "className": "icon right",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-arrow-right right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Next month",
                  "className": "right",
                },
                "ref": null,
                "rendered": "Next month",
                "type": "span",
              },
            ],
            "type": "a",
          },
        ],
        "type": "span",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TimelineTabs
            geographicalWeightData={undefined}
/>,
          <input
            className="progress"
            max={76}
            min="0"
            onChange={undefined}
            type="range"
            value={10}
/>,
          <span
            className="field has-addons has-addons-centered"
>
            <a
                        className="button left"
                        id="previous"
                        onClick={undefined}
            >
                        <span
                                    className="icon left"
                        >
                                    <i
                                                className="fas fa-arrow-left left"
                                    />
                        </span>
                        <span
                                    className="left"
                        >
                                    Previous month
                        </span>
            </a>
            <a
                        className="button right"
                        id="next"
                        onClick={undefined}
            >
                        <span
                                    className="icon right"
                        >
                                    <i
                                                className="fas fa-arrow-right right"
                                    />
                        </span>
                        <span
                                    className="right"
                        >
                                    Next month
                        </span>
            </a>
</span>,
        ],
        "className": "tile is-child notification",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "geographicalWeightData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "progress",
            "max": 76,
            "min": "0",
            "onChange": undefined,
            "type": "range",
            "value": 10,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="button left"
                id="previous"
                onClick={undefined}
>
                <span
                                className="icon left"
                >
                                <i
                                                className="fas fa-arrow-left left"
                                />
                </span>
                <span
                                className="left"
                >
                                Previous month
                </span>
</a>,
              <a
                className="button right"
                id="next"
                onClick={undefined}
>
                <span
                                className="icon right"
                >
                                <i
                                                className="fas fa-arrow-right right"
                                />
                </span>
                <span
                                className="right"
                >
                                Next month
                </span>
</a>,
            ],
            "className": "field has-addons has-addons-centered",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="icon left"
>
                    <i
                                        className="fas fa-arrow-left left"
                    />
</span>,
                  <span
                    className="left"
>
                    Previous month
</span>,
                ],
                "className": "button left",
                "id": "previous",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="fas fa-arrow-left left"
/>,
                    "className": "icon left",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-arrow-left left",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Previous month",
                    "className": "left",
                  },
                  "ref": null,
                  "rendered": "Previous month",
                  "type": "span",
                },
              ],
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="icon right"
>
                    <i
                                        className="fas fa-arrow-right right"
                    />
</span>,
                  <span
                    className="right"
>
                    Next month
</span>,
                ],
                "className": "button right",
                "id": "next",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="fas fa-arrow-right right"
/>,
                    "className": "icon right",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-arrow-right right",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Next month",
                    "className": "right",
                  },
                  "ref": null,
                  "rendered": "Next month",
                  "type": "span",
                },
              ],
              "type": "a",
            },
          ],
          "type": "span",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
